components/automate-cli/cmd/chef-automate/summary.go

```diff
@@ -0,0 +1,590 @@
+					NodeIps:                  chefServerIps,
+					Single:                   false,
+					NodeType:                 true,
+					SkipPrintOutput:          true,
+					HideSSHConnectionMessage: true,
+				},
+			},
+			Opensearch: &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Postgresql: &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Infra:      ss.infra,
+		}
+		err := ss.prepareFEScript(chefServerIps, chefServerNodeMap, "chef-server", "FE")
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(opensearchIps) != 0 {
+		openSearchNodeMap := &NodeTypeAndCmd{
+			Frontend:   &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Automate:   &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			ChefServer: &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Opensearch: &Cmd{
+				CmdInputs: &CmdInputs{
+					Cmd:                      "",
+					NodeIps:                  opensearchIps,
+					Single:                   false,
+					NodeType:                 true,
+					SkipPrintOutput:          true,
+					HideSSHConnectionMessage: true,
+				},
+			},
+			Postgresql: &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Infra:      ss.infra,
+		}
+		err := ss.prepareBEScript(opensearchIps, openSearchNodeMap, "opensearch", "BE")
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(postgresqlIps) != 0 {
+		postgresqlNodeMap := &NodeTypeAndCmd{
+			Frontend:   &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Automate:   &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			ChefServer: &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Opensearch: &Cmd{CmdInputs: &CmdInputs{NodeType: false}},
+			Postgresql: &Cmd{
+				CmdInputs: &CmdInputs{
+					Cmd:                      "",
+					NodeIps:                  postgresqlIps,
+					Single:                   false,
+					NodeType:                 true,
+					SkipPrintOutput:          true,
+					HideSSHConnectionMessage: true,
+				},
+			},
+			Infra: ss.infra,
+		}
+		err := ss.prepareBEScript(postgresqlIps, postgresqlNodeMap, "postgresql", "BE")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ss *Summary) getIPAddressesFromFlagOrInfra() ([]string, []string, []string, []string, *list.List) {
+	errorList := list.New()
+	if ss.statusSummaryCmdFlags.node != "" {
+		automateIps, chefServerIps, opensearchIps, postgresqlIps, errorList := ss.getIpAddressesFromFlag(errorList)
+		return automateIps, chefServerIps, opensearchIps, postgresqlIps, errorList
+	}
+	return ss.getIpAddressesFromInfra(errorList)
+}
+
+func (ss *Summary) getIpAddressesFromFlag(errorList *list.List) ([]string, []string, []string, []string, *list.List) {
+	var automateIps, chefServerIps, opensearchIps, postgresqlIps, nodes []string
+	if ss.statusSummaryCmdFlags.isAutomate ||
+		ss.statusSummaryCmdFlags.isChefServer ||
+		ss.statusSummaryCmdFlags.isOpenSearch ||
+		ss.statusSummaryCmdFlags.isPostgresql {
+		nodes = ss.splitIP(
+			ss.statusSummaryCmdFlags.node,
+		)
+		if len(nodes) != 0 {
+			if ss.statusSummaryCmdFlags.isAutomate {
```

Comment:
We can reduce the inputs to the function as having too many inputs is a code smell

Reply:
Done